plugins/inputs/logfile/logfile.go

```diff
@@ -394,6 +395,25 @@ func (t *LogFile) cleanUpStoppedTailerSrc() {
 	}
 }
 
+func (t *LogFile) editRetentions() {
+	configSet := make(map[string]int)
+	for i := range t.FileConfig {
+		fileconfig := &t.FileConfig[i]
+
+		if fileconfig.LogGroupName != "" {
+			configSet[fileconfig.LogGroupName] += 1
+		}
+
+	}
+	for i := range t.FileConfig {
+		fileconfig := &t.FileConfig[i]
+
+		if fileconfig.LogGroupName != "" && configSet[fileconfig.LogGroupName] > 1 {
+			fileconfig.RetentionInDays = -1
```

Comment:
Can you explain a little why we choose to use -1 to mark Retention to be ignored over 0. Guess that don't make too much of difference but I'm curious if there's any particular reason.

Reply:
I felt like -1 would be more indicative of `never expire` than 0 would. Also I felt like -1 would be a bit less bug prone incase we have any unset values when Retention would be undefined or 0